community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/PlanFingerprint.scala

```diff
@@ -24,20 +24,21 @@ import org.neo4j.helpers.Clock
 
 case class PlanFingerprint(creationTimeMillis: Long, txId: Long, snapshot: GraphStatisticsSnapshot)
 
-case class PlanFingerprintReference(clock: Clock, ttl: Long, statsDivergenceThreshold : Double,
-                                    var fingerprint: Option[PlanFingerprint])
-{
+class PlanFingerprintReference(clock: Clock, minimalTimeToLive: Long, statsDivergenceThreshold : Double,
+                               private var fingerprint: Option[PlanFingerprint]) {
+
   def isStale(lastCommittedTxId: () => Long, statistics: GraphStatistics): Boolean = {
     fingerprint.fold(false) { f =>
-      lazy val currentTimeMillis = clock.currentTimeMillis()
+      val currentTimeMillis = clock.currentTimeMillis()
       lazy val currentTxId = lastCommittedTxId()
 
-      check(f.creationTimeMillis + ttl <= currentTimeMillis, {}) &&
-      check(currentTxId != f.txId, { fingerprint = Some(f.copy(creationTimeMillis = currentTimeMillis)) }) &&
+      f.creationTimeMillis + minimalTimeToLive <= currentTimeMillis &&
+      check(currentTxId != f.txId,
+        () => { fingerprint = Some(f.copy(creationTimeMillis = currentTimeMillis)) }) &&
       check(f.snapshot.diverges(f.snapshot.recompute(statistics), statsDivergenceThreshold),
-            { fingerprint = Some(f.copy(creationTimeMillis = currentTimeMillis, txId = currentTxId)) })
+        () => { fingerprint = Some(f.copy(creationTimeMillis = currentTimeMillis, txId = currentTxId)) })
     }
   }
 
-  private def check(test: => Boolean, ifFalse: => Unit ) = if (test) { true } else { ifFalse ; false }
+  private def check(test: => Boolean, ifFalse: () => Unit ) = if (test) { true } else { ifFalse ; false }
```

Comment:
I think this whole complex statement is not easy to read. I'd prefer if those were individual statements perhaps with a clear variable name to hold the results, and potentially an early return.